saturn-datepicker/src/datepicker/datepicker.ts

```diff
@@ -130,6 +135,66 @@ export class SatDatepickerContent<D> extends _SatDatepickerContentMixinBase
       this.datepicker.close();
     }
   }
+
+  setPreset(preset: string) {
+    switch (preset) {
+      case 'today':
+        const today = moment().toDate();
+
+        this._calendar.beginDate = this.dateAdapter.deserialize(today);
+        this._calendar.endDate = this.dateAdapter.deserialize(today);
+        break;
+      case 'yesterday':
+        const yesterday = moment().subtract('1', 'day').toDate();
+
+        this._calendar.beginDate = this.dateAdapter.deserialize(yesterday);
+        this._calendar.endDate = this.dateAdapter.deserialize(yesterday);
+        break;
+      case 'two_days':
+        const twoDays = moment().subtract('2', 'days').toDate();
+
+        this._calendar.beginDate = this.dateAdapter.deserialize(twoDays);
+        this._calendar.endDate = this.dateAdapter.deserialize(twoDays);
+        break;
+      case 'this_week':
+        const startOfWeek = moment().startOf('week').toDate();
+        const endOfWeek = moment().endOf('week').toDate();
+
+        this._calendar.beginDate = this.dateAdapter.deserialize(startOfWeek);
+        this._calendar.endDate = this.dateAdapter.deserialize(endOfWeek);
+        break;
+      case 'last_week':
+        const startOfPreviousWeek = moment().subtract('1', 'week').startOf('week').toDate();
+        const endOfPreviousWeek = moment().subtract('1', 'week').endOf('week').toDate();
+
+        this._calendar.beginDate = this.dateAdapter.deserialize(startOfPreviousWeek);
+        this._calendar.endDate = this.dateAdapter.deserialize(endOfPreviousWeek);
+        break;
+      case 'this_month':
+        const startOfMonth = moment().startOf('month').toDate();
+        const endOfMonth = moment().endOf('month').toDate();
+
+        this._calendar.beginDate = this.dateAdapter.deserialize(startOfMonth);
+        this._calendar.endDate = this.dateAdapter.deserialize(endOfMonth);
+        break;
+      case 'last_month':
+        const startOfPreviousMonth = moment().subtract('1', 'month').startOf('month').toDate();
+        const endOfPreviousMonth = moment().subtract('1', 'month').endOf('month').toDate();
+
+        this._calendar.beginDate = this.dateAdapter.deserialize(startOfPreviousMonth);
+        this._calendar.endDate = this.dateAdapter.deserialize(endOfPreviousMonth);
+        break;
+    }
+
+    this._calendar.activeDate = this._calendar.beginDate;
+    this._calendar.beginDateSelectedChange.emit(this._calendar.beginDate);
+    this._calendar.dateRangesChange.emit({ begin: this._calendar.beginDate, end: this._calendar.endDate });
+
+    // close with delay, just to show user the selected range
+    setTimeout(() => {
+      this.datepicker.close();
+    }, 200);
+  }
```

Comment:
I'll remove setTimeout

Reply:
yeah, at first it was a good idea but meh, not anymore.